projects/dotcms-field-elements/src/components/dot-date-range/dot-date-range.tsx

```diff
@@ -0,0 +1,168 @@
+    @Event() statusChange: EventEmitter<DotFieldStatusEvent>;
+
+    fp: any;
+
+    /**
+     * Reset properties of the filed, clear value and emit events.
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+        this.emitValueChange();
+    }
+
+    componentWillLoad(): void {
+        this.status = getOriginalStatus(this.isValid());
+        this.emitStatusChange();
+    }
+
+    componentDidLoad(): void {
+        this.fp = flatpickr(`#${this.name}`, {
+            mode: 'range',
+            dateFormat: 'Y-m-d',
+            defaultDate: this.value ? this.value.split(',') : '',
+            maxDate: this.max,
+            minDate: this.min,
+            onChange: this.setValue.bind(this),
+        });
+    }
+
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid(), this.required)
+        };
+    }
+
+    render() {
+        const labelTagParams: DotLabel = {name: this.name, label: this.label, required: this.required};
+        return (
+            <Fragment>
+                <link
+                    rel="stylesheet"
+                    href="https://cdn.jsdelivr.net/npm/flatpickr/dist/flatpickr.min.css"
+                />
+                {getTagLabel(labelTagParams)}
+                <input
+                    class={getErrorClass(this.status.dotValid)}
+                    disabled={this.disabled || null}
+                    id={this.name}
+                    required={this.required || null}
+                    type='text'
+                    value={this.value}
+                />
+                <select onChange={this.setPreset.bind(this)}>
+                    {this.presets.map((item) => {
+                        return (<option value={item.time}>{item.label}</option>);
+                    })}
+                </select>
+                {getTagHint(this.hint)}
+                {getTagError(this.showErrorMessage(), this.getErrorMessage())}
+            </Fragment>
+        );
+    }
+
+    private setPreset(event) {
+        const dateRange = [];
+        const dt = new Date();
+        dt.setDate(dt.getDate() + parseInt(event.target.value, 10));
+
+        if (event.target.value.indexOf('-') > -1) {
+            dateRange.push(dt);
+            dateRange.push(new Date());
+        } else {
+            dateRange.push(new Date());
+            dateRange.push(dt);
+        }
+
+        this.fp.setDate(dateRange, true);
+    }
+
+    private isValid(): boolean {
+        return !(this.required && !(this.value && this.value.length));
+    }
+
+    private setValue(selectedDates, _dateStr, _instance): void {
+        this.value =
+            selectedDates && selectedDates.length === 2
+                ? `${selectedDates[0].toISOString().split('T')[0]},${
+                      selectedDates[1].toISOString().split('T')[0]
+                  }`
+                : '';
```

Comment:
the value not should be a String, we should keep it in a better type and turn it into a String in emitValueChange